File: LeetCode 30 Day Challenge/07-countingElements.js
const countElements = (arr) => {
    arr.sort((a,b) => a - b)
    let count = 0
    let map = {}
    for (const num of arr) {
        if (!map[num]) {
            map[num] = 1
        } else {
            map[num]++
        }
    }
    for (const element of arr) {
        if (map[element + 1]) {
            count++
        }
    }
    return count
}

File: LeetCode 30 Day Challenge/07-counting_elements.py
class Solution:
    def counting_element(self, arr):
        count = 0
        dict = {}
        for num in arr:
            if num in dict:
                dict[num] += 1
            else:
                dict[num] = 1
        for x in arr:
            if x + 1 in dict:
                count += 1
        return count